recipe/build_pytorch.sh

```diff
@@ -128,20 +128,29 @@ if [[ ${cuda_compiler_version} != "None" ]]; then
     export USE_CUDA=1
     if [[ ${cuda_compiler_version} == 9.0* ]]; then
         export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;7.0+PTX"
+        export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
     elif [[ ${cuda_compiler_version} == 9.2* ]]; then
         export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0+PTX"
+        export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
     elif [[ ${cuda_compiler_version} == 10.* ]]; then
         export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5+PTX"
+        export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
     elif [[ ${cuda_compiler_version} == 11.0* ]]; then
         export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0+PTX"
+        export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
     elif [[ ${cuda_compiler_version} == 11.1 ]]; then
         export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
+        export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
     elif [[ ${cuda_compiler_version} == 11.2 ]]; then
         export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
+        export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
     elif [[ ${cuda_compiler_version} == 11.8 ]]; then
         export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9+PTX"
+        export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
     elif [[ ${cuda_compiler_version} == 12.0 ]]; then
         export TORCH_CUDA_ARCH_LIST="5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX"
+        # $CUDA_HOME not set in CUDA 12.0. Using $PREFIX
+        export CUDA_TOOLKIT_ROOT_DIR="${PREFIX}"
```

Comment:
Maybe we need to point PyTorch to the particular target directory. Usually this hasn't been needed with CMake builds, but there is at least one case where we had to go this route ( https://github.com/conda-forge/kaldi-feedstock/pull/39 )

Here's [the layout logic]( https://github.com/conda-forge/cuda-nvcc-feedstock/blob/394f3817decd9316269b5a40e5076b8bcc94dc33/recipe/activate.sh#L5-L7 ) for context

```suggestion
        if [[ "${target_platform}" == "linux-64" ]]; then
            export CUDA_TOOLKIT_ROOT_DIR="${PREFIX}/targets/x86_64-linux"
        elif [[ "${target_platform}" == "linux-ppc64le" ]]; then
            export CUDA_TOOLKIT_ROOT_DIR="${PREFIX}/targets/ppc64le-linux"
        elif [[ "${target_platform}" == "linux-aarch64" ]]; then
            export CUDA_TOOLKIT_ROOT_DIR="${PREFIX}/targets/sbsa-linux"
        else
            echo "Unsupported architecture by CUDA"
            exit 1
        fi
```

Reply:
Also if we would rather just do the `linux-64` logic, that's fine. Just wanted to spell out the other architectures for completeness